test/e2e/csc_tests.go

```diff
@@ -0,0 +1,133 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+const (
+	testResourceName string = "test-operators"
+)
+
+func runCSCWithInvalidTargetNamespace(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables
+	f := test.Global
+	// Run tests
+	if err := cscWithInvalidTargetNamespace(t, f, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func cscWithInvalidTargetNamespace(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Define variables to be used in test
+	targetNamespace := "nonexisting-namespace"
+	packages := "etcd,federationv2,mongodb,amq-streams,couchbase,descheduler,prometheus,svcat,clusterlogging,dynatrace,metering"
+
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// Create the operatorsource to download the manifests
+	testOperatorSource := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-operators",
+			Namespace: namespace,
+		},
+		Spec: operator.OperatorSourceSpec{
+			Type:              "appregistry",
+			Endpoint:          "https://quay.io/cnr",
+			RegistryNamespace: "marketplace_e2e",
+		},
+	}
+	err = createRuntimeObject(f, ctx, testOperatorSource)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig
+	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, testResourceName)
+	if err != nil {
+		return err
+	}
+
+	// Delete the existing catalogsourceconfig
+	err = deleteRuntimeObject(f, resultCatalogSourceConfig, testResourceName, namespace)
```

Comment:
Why are you deleting the datastore `CatalogSourceConfig`?

Reply:
After looking into this I realized that I shouldn't delete this CSC. I can just create the CSC that points to an invalid `targetNamespace`.

This also means that the `deleteRuntimeObject` method in the `helper.go` and so I will remove it.